src/libmeasurement_kit/nettests/runnable.cpp

```diff
@@ -262,28 +262,42 @@ void Runnable::begin(Callback<Error> cb) {
                 }
                 size_t num_entries = 0;
                 if (needs_input) {
-                    if (input_filepath == "") {
-                        logger->warn("an input file is required");
+                    if (input_filepaths.size() <= 0) {
+                        logger->warn("at least an input file is required");
                         cb(MissingRequiredInputFileError());
                         return;
                     }
-                    std::ifstream input_generator{input_filepath};
-                    if (!input_generator.good()) {
-                        logger->warn("cannot read input file");
-                        cb(CannotOpenInputFileError());
-                        return;
-                    }
-
-                    // Count the number of entries
-                    std::string next_input;
-                    while ((std::getline(input_generator, next_input))) {
-                        num_entries += 1;
-                        inputs.push_back(next_input);
+                    std::string probe_cc_lowercase;
+                    for (auto &c: probe_cc) {
+                        /*
+                         * Note: in general the snippet below is not
+                         * so good because it does not work for UTF-8
+                         * and the like but here we are converting
+                         * country codes which are always ASCII.
+                         */
+                        probe_cc_lowercase += std::tolower(c);
                     }
-                    if (!input_generator.eof()) {
-                        logger->warn("cannot read input file");
-                        cb(FileIoError());
-                        return;
+                    for (auto input_filepath: input_filepaths) {
+                        input_filepath = std::regex_replace(
+                            input_filepath,
+                            std::regex{R"(\$\{cc\})"},
+                            probe_cc_lowercase
+                        );
+                        std::ifstream input_generator{input_filepath};
+                        if (!input_generator.good()) {
+                            logger->warn("cannot open input file");
```

Comment:
I understand why you did this here (so that when the country specific list does not exist we don't throw an error), but maybe we should check at the end of this loop if there is at least once good file and if not raise an exception because it means we are actually not going to be able to do much with this test.